pulser-core/pulser/_seq_drawer.py

```diff
@@ -88,12 +103,14 @@ def gather_data(seq: pulser.sequence.Sequence) -> dict:
             time += [time[-1] + 1, total_duration - 1]
             amp += [0, 0]
             detuning += [0, 0]
+            phase += [phase[-1] if len(phase) else 0] * 2
         # Store everything
         time.pop(0)  # Removes the -1 in the beginning
         data[ch] = {
```

Comment:
Maybe a real data structure would improve the code readability. It could additionally integrate some logic. For example:

```python
class ChannelDrawContent:
    time: list[int]
    amp: list[float]
    detuning: list[float]
    phase: list[float]
    target: list[dict[Union[str, tuple[int, int]], Any]]

    def curves(self, draw_phase_curve: bool): ...
```